File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcSpatialElement/IfcSpatialStructureElement/IfcFacilityPart/IfcRailwayPart/IfcRailwayPart.py
class IfcRailwayPart(IfcFacilityPart):
    def __init__(self):
        self.PredefinedType: IfcRailwayPartTypeEnum = None

    # === WHERE CLAUSES ===
    # CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcRailwayPartTypeEnum.USERDEFINED) OR ((PredefinedType = IfcRailwayPartTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))

File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcSpatialElement/IfcSpatialStructureElement/IfcFacilityPart/IfcRailwayPart/IfcRailwayPart.js
class IfcRailwayPart extends IfcFacilityPart {
    constructor() {
        /** @type {IFCRAILWAYPARTTYPEENUM} */
        this.PredefinedType = null;
    }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcRailwayPartTypeEnum.USERDEFINED) OR ((PredefinedType = IfcRailwayPartTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
}
